src/ApiGateways/Mobile.Bff.Shopping/aggregator/Services/CatalogService.cs

```diff
@@ -1,43 +1,41 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Threading.Tasks;
-using Microsoft.eShopOnContainers.BuildingBlocks.Resilience.Http;
+﻿using Microsoft.eShopOnContainers.Mobile.Shopping.HttpAggregator.Config;
+using Microsoft.eShopOnContainers.Mobile.Shopping.HttpAggregator.Models;
 using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Options;
 using Newtonsoft.Json;
-using Microsoft.eShopOnContainers.Mobile.Shopping.HttpAggregator.Config;
-using Microsoft.eShopOnContainers.Mobile.Shopping.HttpAggregator.Models;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading.Tasks;
 
 namespace Microsoft.eShopOnContainers.Mobile.Shopping.HttpAggregator.Services
 {
     public class CatalogService : ICatalogService
     {
-
-        private readonly IHttpClient _apiClient;
+        private readonly HttpClient _httpClient;
         private readonly ILogger<CatalogService> _logger;
         private readonly UrlsConfig _urls;
 
-        public CatalogService(IHttpClient httpClient, ILogger<CatalogService> logger, IOptionsSnapshot<UrlsConfig> config)
+        public CatalogService(HttpClient httpClient, ILogger<CatalogService> logger, IOptions<UrlsConfig> config)
         {
-            _apiClient = httpClient;
+            _httpClient = httpClient;
             _logger = logger;
             _urls = config.Value;
         }
 
         public async Task<CatalogItem> GetCatalogItem(int id)
         {
-            var data = await _apiClient.GetStringAsync(_urls.Catalog + UrlsConfig.CatalogOperations.GetItemById(id));
-            var item = JsonConvert.DeserializeObject<CatalogItem>(data);
-            return item;
+            var stringContent = await _httpClient.GetStringAsync(_urls.Catalog + UrlsConfig.CatalogOperations.GetItemById(id));
+            var catalogItem = JsonConvert.DeserializeObject<CatalogItem>(stringContent);
+
+            return catalogItem;
         }
 
         public async Task<IEnumerable<CatalogItem>> GetCatalogItems(IEnumerable<int> ids)
         {
-            var data = await _apiClient.GetStringAsync(_urls.Catalog + UrlsConfig.CatalogOperations.GetItemsById(ids));
-            var item = JsonConvert.DeserializeObject<CatalogItem[]>(data);
-            return item;
+            var stringContent = await _httpClient.GetStringAsync(_urls.Catalog + UrlsConfig.CatalogOperations.GetItemsById(ids));
```

Comment:
While you're at it, fix these calls by using the GetJsonAsync methods in the https://www.nuget.org/packages/Microsoft.AspNet.WebApi.Client/ package.

Reply:
I cannot use this package in a asp.net core project since it is targeted for ASP.NET.
The best I can do is below:
```var response = await _httpClient.GetAsync(_urls.Basket + UrlsConfig.BasketOperations.GetItemById(id));```
```BasketData basket = response.IsSuccessStatusCode ? await response.Content.ReadAsAsync<BasketData>() : null;```

Reply:
Besides, the author of the project allowed this operations are invoked like that. So I think nothing to do with that. What do you think @CESARDELATORRE